evalml/data_checks/class_imbalance_data_check.py

```diff
@@ -77,6 +78,8 @@ def validate(self, X, y):
         y = _convert_woodwork_types_wrapper(y.to_series())
 
         fold_counts = y.value_counts(normalize=False)
+        if len(fold_counts) == 0:
+            return results
```

Comment:
@bchen1116 in what case does this situation happen? Should this result in an exception, or a data check error, instead of silently continuing?

Reply:
@dsherry This primarily occurs if `y` is empty or if `y` is all null/nan values. One of our tests passes in an empty `y`, so this was added to handle that before we try to index into it.